lib/fastlane/plugin/instabug_official/actions/instabug_official_action.rb

```diff
@@ -5,30 +5,36 @@ module Fastlane
     module Actions
         class InstabugOfficialAction < Action
             def self.run(params)
-            api_token = params[:api_token]
-            
-            if params[:dsym_path].end_with?('.zip')
-            dsym_zip_path = params[:dsym_path].shellescape
-            else
-            dsym_path = params[:dsym_path]
-            dsym_zip_path = ZipAction.run(path: dsym_path).shellescape
-        end
-        
-        endpoint = 'https://api.instabug.com/api/ios/v1/dsym'
-        
-        command =  "curl #{endpoint} --write-out %{http_code} --silent --output /dev/null -F dsym=@\"#{dsym_zip_path}\" -F token=\"#{api_token}\""
-        
-        UI.verbose command
-        
-        return command if Helper.test?
-        
-        result = Actions.sh(command)
-        if result == "200"
-        UI.success 'dSYM is successfully uploaded to Instabug 🤖'
-        else
-        UI.error "Something went wrong during Instabug dSYM upload. Status code is #{result}"
-    end
-end
+                api_token = params[:api_token]
+
+                endpoint = "https://api.instabug.com/api/sdk/v3/symbols_files"
+                command = "curl #{endpoint} --write-out %{http_code} --silent --output /dev/null -F os=\"ios\" -F application_token=\"#{api_token}\" -F symbols_file="
+
+                curlCommand = ""
+                single_path = params[:dsym_path]
+
+                 if single_path != nil
+                    curlCommand += build_single_file_command(command, single_path)
+                 else
+                    dsym_paths = params[:dsym_array_paths].uniq
```

Comment:
First thanks a lot for your contributions🙏. But I have a single question why you did not add then in a zip file and upload it once instead of upload each file separately.

Reply:
To be honest I didn't thought about that 😂 but even if I did, I wouldn't be sure if that even works. Because I'm also not a ruby developer I tried to achieve the result with minimum effort, which was for me do the pipe for calls.